openspp_dms_ftp_server/clients.py

```diff
@@ -35,9 +35,9 @@ def login(self, username: str, password: str) -> bool:
         response = requests.post(
             url=url, json=data, headers={"Content-Type": "application/json"}
         )
-        data = response.json()
-        if not response.status_code == 200:
+        if response.status_code != 200:
             raise OpenSPPClientException(f"Unsuccessful login. Response data: {data}")
+        data = response.json()
         result = response.json().get("result", "")
```

Comment:
Isn't this check redundant as the code is already checking `response.status_code` and raising an exception if it is != 200?

Reply:
@renceInbox I see that the test `test_unsuccessful_authentication` is checking the contents of the JSON response and even if the result is 200, the contents of the JSON seems to indicate that there is still an error. Can you please remind me what the source of this error could be?

Reply:
Hi @kneckinator , this could happen in login. If we provide an invalid username and password the response will be 200 but the result will say 401.